test/e2e/cloud-provider-azure.go

```diff
@@ -60,11 +60,11 @@ func InstallCalicoAndCloudProviderAzureHelmChart(ctx context.Context, input clus
 		options.StringValues = append(options.StringValues, fmt.Sprintf("cloudNodeManager.imageTag=%s", os.Getenv("IMAGE_TAG_CNM")))
 	}
 
-	if input.ConfigCluster.Flavor == "flatcar" {
+	if strings.Contains(input.ClusterName, "flatcar") {
```

Comment:
The new e2e watcher interface uses a different input struct that doesn't have `Flavor`, so the only other way I could think of to detect Flatcar was to look at the cluster name. This is just for e2e, so that will probably work, but feels fragile.

Reply:
That's pretty unfortunate when they remove stuff you need... Maybe we can just add a comment for now documenting what you just described.

Reply:
Uh was that intentional or is it possible it was an accidental omission?

Reply:
It was part of a large test refactoring in kubernetes-sigs/cluster-api#8833 that changed the `Waiter` interface apparently with the assumption it was not being used outside CAPI:

https://github.com/kubernetes-sigs/cluster-api/pull/8833/files#diff-7084fa27f2b10b8d2472295c14187d21ea186aae39efa12e051f340562a4cca4R223

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/9166

Reply:
Q: I don't have the full overview over this could, but could you still use ApplyClusterTemplateAndWaitInput in this func? (I don't see the directly correlation to the Waiter interface, but there is probably an indirect one?)

Reply:
Okay found it. Propagates down from EnsureControlPlaneInitialized